pkg/util/validation/exporter.go

```diff
@@ -0,0 +1,85 @@
+package validation
+
+import (
+	"context"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+const typeTenant = "tenant"
+
+// OverridesExporter exposes per-tenant resource limit overrides as Prometheus metrics
+type OverridesExporter struct {
+	services.Service
+
+	overridesCh  <-chan map[string]*Limits
+	overridesMtx sync.RWMutex
+	overrides    map[string]*Limits
+
+	description *prometheus.Desc
+}
+
+// NewOverridesExporter creates an OverridesExporter that reads updates to per-tenant
+// limits from the provided channel. A single instance of the tenant to limit mapping
+// is kept in memory at a time by this exporter.
+func NewOverridesExporter(overridesCh <-chan map[string]*Limits) *OverridesExporter {
+	exporter := &OverridesExporter{
+		overridesCh:  overridesCh,
+		overridesMtx: sync.RWMutex{},
+
+		// The name and labels for this metric are picked for compatibility with an existing tool
```

Comment:
Totally fine to change the name of this metric or the labels included. Like the comment mentions, this was picked for compatibility with the existing grafana/cortex-tools exporter

Reply:
Will do :+1: